src/devenv/yaml.py

```diff
@@ -0,0 +1,53 @@
+import json
+import sys
+import os
+from pathlib import Path
+
+from strictyaml import Map, MapPattern, Str, Seq
+from strictyaml import load, Bool, Any, Optional, YAMLError, EmptyDict
+
+
+inputsSchema = EmptyDict() | MapPattern(Str(), Map({
+      "url": Str(),
+      Optional("flake", default=None): Bool(),
+      Optional("inputs", default=None): Any(),
+      Optional("overlays", default=None): Seq(Str())
+}))
+
+schema = Map({
+      Optional("inputs", default=None): inputsSchema,
+      Optional("allowUnfree", default=False): Bool(),
+      Optional("imports", default=None): Seq(Str()),
+      Optional("permittedInsecurePackages", default=None): Seq(Str())
+})
+
+YAML_FILE = Path("devenv.yaml")
+
+def read_yaml():
```

Comment:
I'd recommend to use mypy with these settings (mainly strict typing): https://github.com/Mic92/nix-update/blob/e880e386a8d2338473af62c0fbb444c266e51113/pyproject.toml#L11

I found consistently adding function types makes the code more readable and copilot is often good at helping you to find the right types.

Reply:
Also black + ruff are nice to have

Reply:
Fully agreed, I have plans once this stabilizes to add types and use mypy.

Reply:
I consistently got better feedback from pyright over mypy, fwiw.